src/opera_utils/_cslc.py

```diff
@@ -148,20 +174,25 @@ def get_zero_doppler_time(filename: Filename, type_: str = "start") -> datetime:
     ----------
     filename : Filename
         Path to the CSLC product.
+    dataset : str, optional
+        The subdataset to read zero doppler time from
     type_ : str, optional
         Either "start" or "stop", by default "start".
+    datetime_format : str, optional
+        The format of the datetime
 
     Returns
     -------
     str
         Full acquisition time.
     """
-    DATETIME_FORMAT = "%Y-%m-%d %H:%M:%S.%f"
 
     def get_dt(in_str):
-        return datetime.strptime(in_str.decode("utf-8"), DATETIME_FORMAT)
+        return datetime.strptime(in_str.decode("utf-8")[:26], datetime_format)
```

Comment:
why 26? if it's necessary, perhaps a comment why

Reply:
added comments